### PR TITLE
Add IsNotNil and tests for it; possibly close #94

### LINKED ISSUE #94: add qt.IsNonNil
I just ported some tests to quicktest and it went pretty smoothly, but writing `qt.Not(qt.IsNil))` felt really verbose. There’s IsTrue and IsFalse for the same reason, I presume.

### PR DESCRIPTION
**Pull request by gilcrest**
I gave #94 a shot - tried to fit the tests for it into your pattern as well. Thanks again for the great repo.

### CODE REVIEW

File: checker.go [hunk 1]
```diff
@@ -482,6 +482,14 @@ func (c *notChecker) Check(got interface{}, args []interface{}, note func(key st
 	return errors.New("unexpected success")
 }
 
+// IsNonNil returns a Checker negating the IsNil checker. IsNonNil
```
Comment (rogpeppe) [inline checker.go:485]: I think I'd go with `IsNotNil` here (or maybe just `NotNil` which is what the original [gocheck](https://github.com/go-check/check/blob/v1/checkers.go#L152) checker was named).

**Review by rogpeppe — APPROVED** [review]
LGTM with one suggestion for a spelling change. I've been swithering for ages about whether this is worthwhile or not. Oh, one thing, please include the docs in the package-level docs along with all the other checks.

File: checker.go (hunk 1 above)
Comment (frankban) [reply checker.go:485]: +1 for `IsNotNil`

Comment (gilcrest) [reply checker.go]: New commit added to go back to `IsNotNil` - one question though - "please include the docs in the package-level docs along with all the other checks" - not sure what you mean by this? Sorry - should I also put the comment about this function somewhere else?

Comment (gilcrest) [reply checker.go]: I didn't touch the README as I thought it was generated, but I'm happy to add as well if that's what you mean? I was talking to @frankban about helping with the README if you decide to go non-generated.

**Comment by gilcrest** [comment]
"please include the docs in the package-level docs along with all the other checks" - @rogpeppe not sure what you mean by this? Sorry - should I also put the comment about this function somewhere else?

I didn't touch the README as I thought it was generated, but I'm happy to add as well if that's what you mean?

**Comment by gilcrest** [comment]
> "please include the docs in the package-level docs along with all the other checks" - @rogpeppe not sure what you mean by this? Sorry - should I also put the comment about this function somewhere else?
> 
> I didn't touch the README as I thought it was generated, but I'm happy to add as well if that's what you mean?

Nevermind - I think I found it - the `doc.go` file...

**Comment by josharian** [comment]
TIL the word swither. Amusingly it is almost synonymous with dither.

File: doc.go [hunk 2]
```diff
@@ -195,6 +195,10 @@ So it's just fine to check an error like this:
 
     c.Assert(err, qt.IsNil)
 
+IsNotNil
+
+IsNotNil returns a Checker negating the IsNil checker. IsNotNil is the equivalent of qt.Not(qt.IsNil)
```
Comment (frankban) [inline doc.go:200]: Please add the usage example here as well.
And I think we should regenerate the readme, but I can do this later.
Related issue: https://github.com/frankban/quicktest/issues/98

**Review by frankban — APPROVED** [review]
Looks good with some comments. Thanks!

File: doc.go (hunk 2 above)
Comment (gilcrest) [reply doc.go:200]: ok, got it - give me a bit to refactor. I'll try and get to it today, tomorrow at latest. Thanks!

File: checker.go [hunk 3]
```diff
@@ -482,6 +482,14 @@ func (c *notChecker) Check(got interface{}, args []interface{}, note func(key st
 	return errors.New("unexpected success")
 }
 
+// IsNotNil returns a Checker negating the IsNil checker. IsNotNil
+// is the equivalent of qt.Not(qt.IsNil)
+func IsNotNil() Checker {
```
[the remark this answers is not in the document]
Comment (gilcrest) [reply checker.go]: Completed in latest commit.

File: checker_test.go [hunk 4]
```diff
@@ -2639,8 +2639,29 @@ got:
   ~["b", "z", "c", "a"]~
 want:
   []string{"a", "c", "z", "b"}
-`),
-}}
+`)},
```
[the remark this answers is not in the document]
Comment (gilcrest) [reply checker_test.go:2662]: Completed in latest commit.

File: doc.go (hunk 2 above)
Comment (gilcrest) [reply doc.go]: Completed in latest commit.

**Comment by gilcrest** [comment]
One *LAST* thing - @josharian mentioned naming it as `IsNonNil` originally and I wanted to call that out. I do believe `IsNonNil` is more semantically correct than `IsNotNil`. I've been down a few rabbit holes trying to figure out which is correct, but there's really no definitive answer. I see IsNonNil in the standard library and non-nil in the standard library documentation. Happy to change to `IsNonNil` if you prefer, otherwise, let me know if there's anything else you need on this issue!

**Comment by frankban** [comment]
Released as `v1.13.0`.

**Pull request merged**